openfl/component/director/director.py

```diff
@@ -119,15 +125,16 @@ async def wait_experiment(self, envoy_name: str) -> str:
 
         return experiment_name
 
-    def get_dataset_info(self):
+    def get_dataset_info(self) -> Tuple[List, List]:
         """Get dataset info."""
         return self.sample_shape, self.target_shape
 
     def get_registered_shards(self) -> list:  # Why is it here?
         """Get registered shard infos."""
         return [shard_status['shard_info'] for shard_status in self._shard_registry.values()]
 
-    async def stream_metrics(self, experiment_name: str, caller: str):
+    async def stream_metrics(self, experiment_name: str,
+                             caller: str) -> AsyncGenerator[Optional[Union[Dict, None]], Any]:
```

Comment:
Equivalently to:
```python3
    async def stream_metrics(self, experiment_name: str,
                             caller: str) -> AsyncGenerator[Optional[Dict], Any]:
```
I suggest: (according to `Generator[yield_type, send_type, return_type] `)
```suggestion
    async def stream_metrics(self, experiment_name: str,
                             caller: str) -> AsyncGenerator[Optional[Dict], None, SomeDefinedType]:
```
`SomeDefinedType` has to be define, my be just `str`?

Reply:
Based on https://docs.python.org/3/library/typing.html#:~:text=class%20typing.AsyncGenerator,SendType%5D.%20For%20example%3A
I guess there should be `AsyncGenerator[Optional[Dict], None]`